### DISCUSSION TITLE
Difference between `sort` and `sort-by`?

### DISCUSSION THREAD

**Discussion by uvtc** [janet-lang/janet#630] [General]
`sort` takes an optional second `by` argument, and `sort-by` takes a function as its first argument.

What's the difference between `sort` and `sort-by`, other than the arg ordering?

**Comment by andrewchambers** [comment]
For sort 'by is an alternative comparison function to use instead of '< for sorting, for sort-by '< is always used for comparison, but first 'by is called on each item.

**Reply by uvtc** [reply]
Thanks for the reply, @andrewchambers . Sorry for the delay in getting back to this.

Ok, I understand. The function you pass as the 2nd arg to `sort` must be able to take two args and compare them.
Whereas, the function you pass to `sort-by` just takes one arg: the object you look at to get the value to sort on.

**Reply by uvtc** [reply]
Added an example to <https://janetdocs.com/sort>.
